File: app/src/main/java/com/zzzmode/appopsx/ui/main/backup/ExportFragment.java
package com.zzzmode.appopsx.ui.main.backup;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.collection.SparseArrayCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.zzzmode.appopsx.R;
import com.zzzmode.appopsx.ui.analytics.AEvent;
import com.zzzmode.appopsx.ui.analytics.ATracker;
import com.zzzmode.appopsx.ui.model.AppInfo;
import java.util.ArrayList;

/**
 * 导出配置
 * Created by zl on 2017/5/7.
 */

public class ExportFragment extends BaseConfigFragment implements View.OnClickListener {

  private static final String TAG = "ExportFragment";

  private ExportAdapter adapter;

  private ConfigPresenter mPresenter;

  @Nullable
  @Override
  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    return inflater.inflate(R.layout.fragment_export, container, false);
  }

  @Override
  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);

    ArrayList<AppInfo> appInfos = getArguments().getParcelableArrayList(BackupActivity.EXTRA_APPS);
    adapter = new ExportAdapter(appInfos);
    recyclerView.setAdapter(adapter);

    view.findViewById(R.id.fab).setOnClickListener(this);

    mPresenter = new ConfigPresenter(getContext().getApplicationContext(), this);
  }

  @Override
  public void onClick(View v) {
    if (v.getId() == R.id.fab) {
      //导出
      export();
      ATracker.send(AEvent.A_EXPORT);
    }
  }


  private void export() {
    SparseArrayCompat<AppInfo> checkedApps = adapter.getCheckedApps();
    final int size = checkedApps.size();
    AppInfo[] appInfos = new AppInfo[size];
    for (int i = 0; i < size; i++) {
      appInfos[i] = checkedApps.valueAt(i);
    }

    mPresenter.export(appInfos);
  }

}

File: app/src/main/kotlin/com/zzzmode/appopsx/ui/main/backup/ExportFragment.kt
package com.zzzmode.appopsx.ui.main.backup

import android.os.Bundle
import android.support.v4.util.SparseArrayCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.zzzmode.appopsx.R
import com.zzzmode.appopsx.ui.analytics.AEvent
import com.zzzmode.appopsx.ui.analytics.ATracker
import com.zzzmode.appopsx.ui.model.AppInfo
import java.util.ArrayList

/**
 * 导出配置
 * Created by zl on 2017/5/7.
 */

class ExportFragment : BaseConfigFragment(), View.OnClickListener {

    private lateinit var adapter: ExportAdapter

    private lateinit var mPresenter: ConfigPresenter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_export, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        arguments?.getParcelableArrayList<AppInfo>(BackupActivity.EXTRA_APPS)?.apply {
            adapter = ExportAdapter(this)
        }

        recyclerView?.adapter = adapter

        view.findViewById<View>(R.id.fab).setOnClickListener(this)

        mPresenter = ConfigPresenter(view.context, this)
    }

    override fun onClick(v: View) {
        if (v.id == R.id.fab) {
            //导出
            export()
            ATracker.send(AEvent.A_EXPORT)
        }
    }


    private fun export() {

        adapter.checkedApps.apply {
            val size = size()

            val appInfos = (0 until size).mapTo(ArrayList<AppInfo>()) { valueAt(it) }

            mPresenter.export(appInfos.toTypedArray())
        }

    }

    companion object {

        private const val TAG = "ExportFragment"
    }

}
